### DISCUSSION TITLE
Batch requests in ISR

### DISCUSSION THREAD

**Discussion by knenkne** [vercel/next.js#36424] [Ideas]
### Describe the feature you'd like to request

Hey!
So, by now we've to return pages slugs from `getStaticPaths`. e.g `[ { params: { id: 1 } }, { params: { id: 2 } } ]` 
And that means `getStaticProps` will do a requests 1 by 1 to some `/endpoint/[id]`
But in my case I've over `500` pages and I don't wanna DDOS `3rd party API` at build time
Therefore such `APIs` usually have batching endpoints like `getMultipleIds`, so in this case we can already grab all data inside of `getStaticPaths` and don't request it 1 by 1 in each `getStaticProps`. 

### Describe the solution you'd like

`getStaticPaths` exports not only `slugs`, but also a `data` for these `slugs`. e.g `[ { params: { id: 1, data: [] } } ]`

### Describe alternatives you've considered

By now I've to create own decorator, that will wait for all `ids` to come in

**Comment by Alexanderdunlop** [comment]
Any update on this?

**Reply by knenkne** [reply]
> Any update on this?

Hey! Unfortunately nothing from Next.js team yet. 
So now I'm hacking it with fetching inside `getStaticPaths` and put it into some `Cache`. Then inside `getStaticProps` I check if there is some data in `Cache` and take it or fetch it
